src/utils.rs

```diff
@@ -0,0 +1,102 @@
+use std::collections::HashMap;
+
+use scale_info::{form::PortableForm, Field, PortableRegistry, Type, TypeDef, TypeParameter};
+
+/// Checks if the two types are equal, but with a bit of a relaxed condition:
+/// If there is any type id involved that is not equal:
```

Comment:
Can you rephrase this and the segment below?

I find it a little hard to read.

> Checks if the two types are equal with the relaxed condition that
if the generic type params are equal then the type id's does not need to by equal